src/utils/ucc_status.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "ucc/api/ucc_status.h"
+#include <stdio.h>
+
+const char *ucc_status_string(ucc_status_t status)
+{
+    static char error_str[128] = {0};
+
+    switch (status) {
+    case UCC_OK:
+        return "Success";
+    case UCC_INPROGRESS:
+        return "Operation is posted and is in progress";
+    case UCC_OPERATION_INITIALIZED:
+        return "Operation initialized but not posted";
+    case UCC_ERR_NOT_SUPPORTED:
+        return "Operation is not supported";
+    case UCC_ERR_NOT_IMPLEMENTED:
+        return "Function is not implemented";
+    case UCC_ERR_INVALID_PARAM:
+        return "Invalid parameter";
+    case UCC_ERR_NO_MEMORY:
+        return "Out of memory";
+    case UCC_ERR_NO_RESOURCE:
+        return "No resources are available to initiate the operation";
+    case UCC_ERR_NO_MESSAGE:
+        return "Unhandled error";
+    case UCC_ERR_NOT_FOUND:
+        return "Component is not found";
```

Comment:
Is it necessarily "component" that is not found?

Reply:
right now we use it only during component load, do you suggest to return just "not found"?

Reply:
i think so, we print the actual error message anyways